src/js/components/DataTable/buildState.js

```diff
@@ -84,11 +84,26 @@ export const filterAndSortData = (data, filters, onSearch, sort) => {
   if (sort && !sort.external) {
     const { property, direction } = sort;
     result = result === data ? [...data] : result; // don't sort caller's data
-    const before = direction === 'asc' ? 1 : -1;
-    const after = direction === 'asc' ? -1 : 1;
+    const sortAsc = direction === 'asc';
+    const before = sortAsc ? 1 : -1;
+    const after = sortAsc ? -1 : 1;
     result.sort((d1, d2) => {
-      if (datumValue(d1, property) > datumValue(d2, property)) return before;
-      if (datumValue(d1, property) < datumValue(d2, property)) return after;
+      const d1Val = datumValue(d1, property);
+      const d2Val = datumValue(d2, property);
+      const d1IsString = typeof d1Val === 'string';
+      const d2IsString = typeof d2Val === 'string';
+      if (d1IsString && d2IsString) {
+        if (sortAsc) {
```

Comment:
Thanks Samuel. This helps. I'd also consider simplifying to something like:
```es6
if (typeof d1Val === 'string' && typeof d2Val === 'string') {
  const result = d1Val.localeCompare(d2Val, undefined, { sensitivity: 'base' });
  return sortAsc ? result : -result;
}
```
It might be a little confusing but entertaining to do
```es6
return before * d1Val.localeCompare(d2Val, undefined, { sensitivity: 'base' });
```
:)

Reply:
Hey @MikeKingdom, I'm good with the former suggestion, but I agree: the latter option might be a little confusing, so I'd prefer to avoid it. I've pushed the change. Thanks!